api-report/container-definitions.api.md

```diff
@@ -246,9 +246,13 @@ export interface IDeltaHandlerStrategy {
 }
 
 // @public
-export interface IDeltaManager<T, U> extends IEventProvider<IDeltaManagerEvents>, IDeltaSender, IDisposable {
+export interface IDeltaManager<T, U> extends IEventProvider<IDeltaManagerEvents>, IDeltaSender {
     readonly active: boolean;
     readonly clientDetails: IClientDetails;
+    // (undocumented)
+    dispose(error?: ICriticalContainerError): void;
```

Comment:
should this even be on the interface? My guess is the only caller is in container, which has the concrete class. I think it could be an implementation detail of the concrete class, and not be exposed to other consumers which consume via interface.

Reply:
Removing it from the interface would cause a sweeping breaking change (since DM is passed around a lot). Probably makes sense to be in its own PR

Reply:
> As interface consumers should never call dispose

@anthony-murphy - I think we do expect this to be used by external consumers.  The app can make the determination whether it's "done done" with the Container and disposes (e.g. Teams switching chats and unloading all Loops) v. just wanting to close to prevent further interaction but willing to leave a readonly view up for the user (e.g. when closing due to an error).

Reply:
Ugh nevermind.  I responded on the main PR page and jumped to a conclusion - I thought this was about `Container` not `DeltaManager`

Reply:
> Probably makes sense to be in its own PR

I'd say do that PR first, if we're going to do it.  It's been this way for at least 3 years so maybe just punt. Not sure the cost, if it's not hard maybe just go for it.

Reply:
Not really sure why it was ever on the interface, since it used to just throw... Feels worth it to me clean up, we are trying to keep our API surface tight.

Reply:
I moved this PR to be in `main` since there isn't really a breaking change. I'll make a follow-up PR in `next` that will remove `dispose` from the interface

Reply:
what are you all talking about, this is literally a change to the definition of the interface, just don't make this change, the other changes are fine

Reply:
this change isn't necessary, just don't modify the interfaces. the concrete class can still add the optional member